File: temas/delegacion/orquesta.cpp
#include "iostream"
#include <vector>
#include "instrumento.cpp"
using namespace std;

class Orquesta{
    public:
        void addInstrumento(Instrumento *i) { Instrumentos.push_back(i); }
        void tocarInstrumentos(){
            for(auto it : Instrumentos){
                it->tocar();
            }
        }
        void afinarInstrumentos(){
            for(auto it : Instrumentos){
                it->afinar();
            }
        }
    private:
        vector<Instrumento*> Instrumentos;
};

File: temas/delegacion/python/orquesta.py
from instrumento import *


class Orquesta:
	def __init__(self):
		self.__instrumentos = []

	def addInstrumento(self, instrumento):
		self.__instrumentos.append(instrumento)
	def tocarInstrumento(self):
		for i in range(len(self.__instrumentos)):
			self.__instrumentos[i].tocar()
	def afinarInstrumento(self):
		for j in range(len(self.__instrumentos)):
			self.__instrumentos[j].afinar()
